native/machine-coding/Weather App/index.js

```diff
@@ -0,0 +1,48 @@
+// API TO BE USED FOR THE WEATHER DETAILS
+const apiKey = "46d47581a51a79782741111953e700af";        
+const apiUrl = "https://api.openweathermap.org/data/2.5/weather?units=metric&q=";
+
+const searchBox = document.querySelector(".search input");
+const searchBtn = document.querySelector(".search button");
+const weatherIcon = document.querySelector(".weather-icon");
+
+async function checkWeather(city){
+    const response = await fetch(apiUrl + city + `&appid=${apiKey}`);   //to fetch the details when city name is entered.
+
+    if(response.status == 404){                                         // when city name is entered wrong
+        document.querySelector(".error").style.display = "block";
+        document.querySelector(".weather").style.display = "none";
+    }
+
+    else{
+    var data = await response.json();
+   
+    document.querySelector(".city").innerHTML = data.name;
+    document.querySelector(".temp").innerHTML = Math.round(data.main.temp) + "°C";
+    document.querySelector(".humidity").innerHTML = data.main.humidity + "%";
+    document.querySelector(".wind").innerHTML = data.wind.speed + "km/h";

```

Comment:
does it work if we keep the selectors outside the function?

Reply:
no its not properly functional like that..........